### ISSUE TITLE
`Rails/Delegate` Cop registers when method uses safety navigator 

### ISSUE THREAD

**Issue by garettarrowood** [bbatsov/rubocop#4991]
`Rails/Delegate` Cop reports an offense when the use of a safety navigator disqualifies the use of delegation.

In the example below, if `Foo.new.bar == nil` and the fix is used, a NilClass error will occur. 

--------

## Expected behavior

I expect the example below to not throw a `Rails/Delegate` offense and also not autocorrect. 

```ruby
class Foo < ApplicationRecord
  def count
    bar&.count
  end
end
```

## Actual behavior

Running `rubocop app/models/foo.rb` on this example returns:

```shell
Offenses:

app/models/foo.rb:2:3: C: Use delegate to define delegations.
  def count
  ^^^
```

Running `rubocop app/models/foo.rb --auto-correct` on this example returns:

```ruby
class Foo < ApplicationRecord
  delegate :count, to: :bar
end
```

## Steps to reproduce the problem

Please use the example above to reproduce. This was generated using Rails 5.0.6.

## RuboCop version

```
[Fri Nov 03 07:55:29]$ rubocop -V
0.49.1 (using Parser 2.4.0.0, running on ruby 2.3.3 x86_64-darwin15)
```

**Comment by Drenmi** [comment]
You can prevent a `NoMethodError` by using `allow_nil: true`:

```ruby
class Foo < ApplicationRecord
  delegate :count, to: :bar, allow_nil: true
end
```

I'll label this as a bug, as the autocorrect should handle that. 🙂

**Comment by garettarrowood** [comment]
Thanks!  Makes sense.

**Comment by Drenmi** [comment]
@michniewicz: Considering that comment, we most likely do not. Closing this as it is fixed in 0.51. 🙂

**Comment by Darhazer** [comment]
It seems whiny_nils was removed from Rails. Can we make Rails/Delegate handle this now? (realising this is now part of rubocop-rails, so the issue should be moved there)?
